### PR TITLE
Fix enabling new CI Build.

### PR DESCRIPTION
**Pull request by HelenaEksler**
@amitaibu Fixed bug when enabling new CI Build.

### CODE REVIEW

File: shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php [hunk 1]
```diff
@@ -91,10 +91,33 @@ public function patchEntity($entity_id) {
   protected function checkEntityAccess($op, $entity_type, $entity) {
     $account = $this->getAccount();
 
-    $wrapper = entity_metadata_wrapper('node', $entity);
-    $repo_name = $wrapper->og_repo->label();
+    if ($op == 'create' && $entity->is_new) {
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php]: `!empty($entity->is_new)`

File: shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php [hunk 2]
```diff
@@ -91,10 +91,33 @@ public function patchEntity($entity_id) {
...
+      // New CI Build is to be created.
+      $request = $this->getRequest();
+      if (!isset($request['repository'])) {
+        // Repository is not send.
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php]: => `Repository is not provided.` -- can this happen - isn't `repository` `required`?

File: shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php [hunk 3]
```diff
@@ -91,10 +91,33 @@ public function patchEntity($entity_id) {
...
+        return FALSE;
+      }
+
+      // Get repository from the request.
+      $repo_id = $request['repository'];
+      $repo = node_load($repo_id);
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php:100]: what if node isn't valid? (i.e. not loaded, or user doesn't have access to it?)

File: shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php (hunk 2 above)
Comment (HelenaEksler) [inline shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php]: https://github.com/shoov/shoov/blob/new-build-create-bug/shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php#L25 - No, it's not required.

**Pull request merged**